### ISSUE TITLE
Biweekly events can end up in the wrong week if start date of the `RRule` and the `RRule` set differ

### ISSUE THREAD

**Issue by manuelVo** [fmeringdal/rust-rrule#87]
I've written a test-case that triggers this bug:

```rust
#[test]
fn weekly_with_interval_3() {
    let rrule_dt_start = ymd_hms(2022, 10, 19, 18, 30, 0);

    let mut rrule = "FREQ=WEEKLY;INTERVAL=2"
        .parse::<RRule<Unvalidated>>()
        .unwrap();
    rrule.count = Some(2);

    let rrule = rrule.validate(rrule_dt_start).unwrap();

    let rrule_set_dt_start = ymd_hms(2022, 10, 24, 18, 30, 0);
    let set = RRuleSet::new(rrule_set_dt_start).rrule(rrule);

    test_recurring_rrule_set(
        set,
        &[
            ymd_hms(2022, 11, 2, 18, 30, 0),
            ymd_hms(2022, 11, 16, 18, 30, 0),
        ],
    );
}
```

The `RRuleSet` should report the dates 2022-11-02 and 2022-11-16 as the dates of the next events. However, it reports 2022-10-26 and 2022-11-09, which is offset one week by where the events should be.

**Comment by fmeringdal** [comment]
One should only use one specific start date when working with a `RRuleSet`, meaning that the date used in `rrule.validate` and `RRuleSet::new` should be the same. We should really try to make the API design of the library better to make it impossible to use multiple start dates for an `RRuleSet`.

**Comment by manuelVo** [comment]
Is there a way of retrieving the next event of a rrule starting from a given DateTime other than iterating over all the past events? That's what I was trying to accomplish with this code.

Btw. thanks for making this library. It saved me a lotnof headache.

**Comment by fmeringdal** [comment]
Yeah, there is `RRuleSet::after` and `RRuleSet::before` to support those use-cases.
So your code should be:
```rust
#[test]
fn weekly_with_interval_3() {
    let rrule_dt_start = ymd_hms(2022, 10, 19, 18, 30, 0);

    let mut rrule = "FREQ=WEEKLY;INTERVAL=2"
        .parse::<RRule<Unvalidated>>()
        .unwrap();
    rrule.count = Some(2);

    let rrule = rrule.validate(rrule_dt_start).unwrap();

    let rrule_set_dt_start = ymd_hms(2022, 10, 24, 18, 30, 0);
    let set = RRuleSet::new(rrule_dt_start).after(rrule_set_dt_start).rrule(rrule);

    test_recurring_rrule_set(
        set,
        &[
            ymd_hms(2022, 11, 2, 18, 30, 0),
            ymd_hms(2022, 11, 16, 18, 30, 0),
        ],
    );
}
```

**Comment by manuelVo** [comment]
I was ignoring `after` so far, since I'm accessing the generated times through the Iterator, where `after` doesn't work. However, I think I may be able to modify my code to use the `all` endpoint instead, though it's not a perfect fit.

Thanks for you assistance and feel free to close the issue.

**Comment by fmeringdal** [comment]
I will take a look into how we can design the API to ensure that only one start date is being used. Thanks for reporting, will close this issue for now.
